invenio/modules/documents/utils.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Document utils."""
+
+from flask import abort
+
+__all__ = ('identifier_to_path', 'identifier_to_path_and_permissions')
+
+
+def identifier_to_path(identifier):
+    """Convert the identifier to path.
+
+    :param str identifier: A unique file identifier
+    :raises NotFound: if the file fullpath is empty
+    """
+    fullpath = identifier_to_path_and_permissions(identifier, path_only=True)
+    return fullpath or abort(404)
+
+
+def identifier_to_path_and_permissions(identifier, path_only=False):
+    """Convert the identifier to path.
+
+    :param str identifier: A unique file identifier
+    :raises Forbidden: if the user doesn't have permissions
+    """
+    if identifier.startswith('recid:'):
+        record_id, filename = _parse_legacy_syntax(identifier)
+
+        fullpath, permissions = _get_legacy_bibdoc(
+            record_id, filename=filename
+        )
+    else:
+        fullpath, permissions = _get_document(identifier)
+
+    if path_only:
+        return fullpath
+
+    try:
+        assert (permissions[0] == 0)
+    except AssertionError:
+        return abort(403)
+    else:
+        return
+
+
+def _get_document(uuid):
+    """Get the document fullpath.
+
+    :param str uuid: The document's uuid
+    """
+    from invenio.modules.documents.api import Document
+    from invenio.modules.documents.errors import (
+        DocumentNotFound, DeletedDocument
+    )
+
+    try:
+        document = Document.get_document(uuid)
+    except (DocumentNotFound, DeletedDocument):
+        path = _simulate_file_not_found()
+    else:
+        path = document.get('uri', ''), document.is_authorized()
```

Comment:
Please note that this will work only for local filesystem. We should assert the opener type until you update the *Flask-IIIF* to accept bytestream.